### ISSUE TITLE
Incorrect use of std::sqrt, std::pow, std::abs.

### ISSUE THREAD

**Issue by cosurgi** [CGAL/cgal#4527]
## Issue Details

I have implemented in [YADE](https://yade-dem.org/doc/) support for [high precision calculations](https://yade-dev.gitlab.io/-/trunk/-/jobs/438538107/artifacts/install/share/doc/yade-ci/html/HighPrecisionReal.html). The problem is that I had to pollute `std::` namespace with this:

```CPP
namespace std {
using ::yade::math::pow;
using ::yade::math::sqrt;
using ::yade::math::abs; // This one is actually written elsewhere in the code, in specific function calls.
}
```

See the file [CgalNumTraits.hpp](https://gitlab.com/yade-dev/trunk/-/blob/master/lib/high-precision/CgalNumTraits.hpp#L161). Best programming practices require that mathematical functions are always called unqualified and let [ADL](https://en.cppreference.com/w/cpp/language/adl) take care of this. See a discussion about this problem in these references (search for word "unqualified"):

1. [boost float128](https://www.boost.org/doc/libs/1_72_0/libs/math/doc/html/math_toolkit/float128_hints.html)
2. [boost multiprecision](https://www.boost.org/doc/libs/1_60_0/libs/multiprecision/doc/html/boost_multiprecision/tut/floats/fp_eg/jel.html)
3. [boost real concepts](https://www.boost.org/doc/libs/1_72_0/libs/math/doc/html/math_toolkit/real_concepts.html)

You can write `using std::pow` or `using std::abs` inside a function which is going to call this. But write the function call itself unqualified, so that ADL will work.

There are several places in CGAL that use `std::sqrt`, for example `PCA_util_Eigen.h` or `
centroid.h`. Maybe you should be using [Sqrt function](https://gitlab.com/yade-dev/trunk/-/blob/master/lib/high-precision/CgalNumTraits.hpp#L75) from numerical traits?

The same applies to `std::pow` and `std::abs` functions. Launching command in CGAL sources:

```sh
grep -E "std::pow|std::sqrt|std::abs" . -r --color
grep -E "=.*std::pow|=.*std::sqrt|=.*std::abs" . -r --color
```

has a lots of hits.

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): linux, see [gitlab-CI pipeline](https://gitlab.com/yade-dev/trunk/-/merge_requests/383/pipelines).
* Compiler: g++, clang
* Release or debug mode: both
* Specific flags used (if any): ` -Werror -Wformat -Wformat-security -Wformat-nonliteral -Wall -Wextra -Wnarrowing -Wreturn-type -Wuninitialized -Wfloat-conversion -Wcast-align -Wdisabled-optimization -Wtrampolines -Wpointer-arith -Wswitch-bool -Wwrite-strings -Wnon-virtual-dtor -Wreturn-local-addr -Wno-error=cpp`
* CGAL version: 4.13
* Boost version: 1.67.0
* Other libraries versions if used (Eigen, TBB, etc.): that's for one particular run. Gitlab pipeline covers more versions.

| library       | cmake                | C++                |
| ------------- | -------------------- | ------------------ |
| boost         | 106700               | 1.67.0             |
| cgal          |                      | 4.13               |
| clp           | 1.16.11              | 1.16.11            |
| cmake         | 3.13.4               |                    |
| coinutils     | 2.10.14              | 2.10.14            |
| compiler      | /usr/bin/g++ 8.3.0   | gcc 8.3.0          |
| eigen         | 3.3.7                | 3.3.7              |
| freeglut      | 2.8.1                |                    |
| gl            |                      | 20180725           |
| ipython       | 5.8.0                |                    |
| metis         |                      | 5.1.0              |
| mpi           | 3.1                  | ompi:3.1.3         |
| mpi4py        | 2.0.0                |                    |
| openblas      |                      |  OpenBLAS 0.3.5    |
| python        | 3.7.3                | 3.7.3              |
| qglviewer     |                      | 2.6.3              |
| qt            |                      | 5.11.3             |
| sphinx        | 1.8.4-final-0        |                    |
| sqlite        |                      | 3.27.2             |
| suitesparse   | 5.4.0                | 5.4.0              |
| vtk           | 6.3.0                | 6.3.0              |

Linux version: Devuan GNU/Linux 3 (beowulf)

**EDIT:** You may want to have a look at Eigen, I reported there the [same bug](https://gitlab.com/libeigen/eigen/issues/1823) but they had this problem only in four lines in entire codebase.

**Comment by mglisse** [comment]
A few places should be changed, but many have arguments that are explicitly double, in which case std::sqrt is fine: `std::pow(10.0, 300.0)`, `std::sqrt(3.)`, `std::sqrt(CGAL::to_double(...))`, etc.

**Comment by cosurgi** [comment]
Ah, I'm sorry. I didn't realize that! At first I wanted to simply make a patch that will fix YADE compilation. But then in the same file I saw plenty of other invocations which my compilation didn't use (or trigger). So I thought that's it's better to report this, to figure out the best approach here. Turns out it was better to ask first ;)

I can prepare a pull request which will fix compiling YADE. It will be a step in the right direction.

Thank you for your quick response! :)
